packages/discovery-provider/plugins/pedalboard/apps/relay/src/middleware/contextInjector.ts

```diff
@@ -1,13 +1,19 @@
-import { Users } from '@pedalboard/storage'
+import { DeveloperApps, Users } from '@pedalboard/storage'
 import { ValidatedRelayRequest } from '../types/relay'
 
 /** Context built by the context injector that can be referenced later in the request (via response object). */
 export interface RequestContext {
   startTime: Date
-  recoveredSigner: Users
   ip: string
   requestId: string
   validatedRelayRequest: ValidatedRelayRequest
+
+  // using these booleans is clearer than
+  // writing a js property validator
+  recoveredSigner: Users | DeveloperApps
+  signerIsApp: boolean
+  signerIsUser: boolean
+  createOrDeactivate: boolean
```

Comment:
these boolean flags are getting a little out of hand, would like to refactor these to something more ergonomic in the future